extensions/notebook/src/book/bookTocManager.ts

```diff
@@ -200,13 +200,13 @@ export class BookTocManager implements IBookTocManager {
 			// if addSection is undefined then we remove the whole section from the table of contents.
 			return addSection;
 		} else {
-			let newSection = convertTo(version, section);
+			let newSection = { ...section };
```

Comment:
Be careful here - because you're not doing a deep clone then objects such as arrays are not going to be copied and so both `section` and `newSection` will refer to the same object.

In the line below you're overwriting the sections so at least that part is handled here but there's also the `chapters` property which is a string array that will have this issue. And in general I'd strongly suggest changing this to be a deep clone here anyways because otherwise if an object property is added later on it'll just silently start doing the wrong thing.